### DISCUSSION TITLE
Error running migrations due to duplicate column name `device_type`

### DISCUSSION THREAD

**Discussion by oSumAtrIX** [dani-garcia/vaultwarden#5058] [Q&A] [answered]
### Vaultwarden Support String

Vaultwarden can not be started, so no support string can be retrieved.

### Vaultwarden Build Version

v1.32.1

### Deployment method

Official Container Image

### Reverse Proxy

nginx

### Host/Server Operating System

Linux

### Operating System Version

Ubuntu 24

### Clients

Web Vault, Browser Extension

### Steps To Reproduce

This is tricky. I reverted to the v1.32.0 image, and the issue disappeared. I was able to access the web vault. Upgrading to the latest gave the same panic. It is unknown how to reproduce this.

### Expected Result

The migration succeeds.

### Actual Result

The migration fails due to a panic.

### Logs

```text
[2024-10-06 22:25:58.407][panic][ERROR] thread 'main' panicked at 'Error running migrations: QueryError(DieselMigrationName { name: "2024-09-04-091351_use_device_type_for_mails", version: MigrationVersion("20240904091351") }, DatabaseError(Unknown, "Duplicate column name 'device_type'"))': src/db/mod.rs:479
   0: vaultwarden::init_logging::{{closure}}
   1: std::panicking::rust_panic_with_hook
   2: std::panicking::begin_panic_handler::{{closure}}
   3: std::sys::backtrace::__rust_end_short_backtrace
   4: rust_begin_unwind
   5: core::panicking::panic_fmt
   6: core::result::unwrap_failed
   7: vaultwarden::db::DbPool::from_config
   8: vaultwarden::main::{{closure}}
   9: vaultwarden::main
  10: std::sys::backtrace::__rust_begin_short_backtrace
  11: main
  12: <unknown>
  13: __libc_start_main
  14: _start
```

### Screenshots or Videos

![image](https://github.com/user-attachments/assets/06cb9e16-0b7a-4db5-b4bb-8f2c8361dfff)

**Comment by BlackDex** [comment, marked as answer]
Looks like something didn't door completed, or you used the testing tagged images also recently.

The only way to fix this correctly is by removing the extra column `device_type` from the `twofactor_incomplete` tabel, and let Vaultwarden run the migration again.

**Reply by oSumAtrIX** [reply]
I have not used any container except for the one tagged with "latest". I would assume those are stable and not test containers. I'll do the manual change in the database

**Reply by oSumAtrIX** [reply]
I am not seeing any extra column here but deleting it solved the issue.

![image](https://github.com/user-attachments/assets/9dc9d029-c4c2-4bd5-a714-5956886874de)

**Reply by BlackDex** [reply]
It wasn't a column named `extra`, but `device_type`, as the error message also indicated.

**Reply by oSumAtrIX** [reply]
Sorry for the confusion, but you said:

> by removing the extra column device_type

I was referring to that column. There is no extra column, only one "device_type" column. The issue is resolved by deleting that column, but doing manual edits like this could be dangerous.

**Reply by BlackDex** [reply]
True, that is why the migration of Diesel handles all this, but sometimes a server does strange things, or loose connection or whatever and then stuff doesn't get stirred. Probably almost never happens. But that is not something we can fix.
